### PR TITLE
samples: lsm6dsl: Add a harness_config

### PR DESCRIPTION
**Pull request by erwango**
In order to enable sample to be run and evaluated in sanitycheck,
add a harness_config to validate sample output.
Tested on disco_l475_iot1

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: samples/sensor/lsm6dsl/sample.yaml [hunk 1]
```diff
@@ -5,3 +5,9 @@ tests:
     harness: console
     tags: sensors
     depends_on: lsm6dsl
+    harness_config:
+      type: multi_line
```
Comment (avisconti) [inline samples/sensor/lsm6dsl/sample.yaml]: Why not using the 'ordered' rule?
ordered: true

Maybe it does not add a big added value, but I would use it.
What do you think?

**Review by avisconti — CHANGES_REQUESTED** [review]
LGTM, but I was wondering if 'ordered' is worth to be used or not.
Anyway, I learnt something new which I might use to improve my samples

File: samples/sensor/lsm6dsl/sample.yaml (hunk 1 above)
Comment (erwango) [reply samples/sensor/lsm6dsl/sample.yaml]: Looking into it. So far, I don't manage to get test passing. Seems like regex not good enough.

**Comment by erwango** [comment]
Setting as DNM since I can't getting it passed.

**Comment by avisconti** [comment]
> Setting as DNM since I can't getting it passed.

Do you mean that test is not passing if you use 'ordered'? Maybe we can just skip it for the moment. What do you think?

**Comment by erwango** [comment]
@avisconti, thanks!
I took me few hours to fix this but it was useful:
Sanitycheck regex does not play well with parenthesis, so I had to rework the harness.
Besides, since sample is running a while loop, test verdict could be done on console output acquired be fore flash completion, providing incoherent verdict.
Using "ordered: true" prevents this and is mandatory for this kind of tests.
I'm not sure how/why I initially saw the test PASSED despite the parenthesis issue., anyway, this is working fine now.

**Review by avisconti — APPROVED** [review]
OK, very good!!

**Pull request merged**